Project-21-React-Animation/src/components/ChallengeItem.jsx

```diff
@@ -0,0 +1,93 @@
+import { useContext } from "react";
+import PropTypes from "prop-types";
+import { AnimatePresence, motion } from "framer-motion";
+
+import { ChallengesContext } from "../store/challenges-context.jsx";
+
+export default function ChallengeItem({
+  challenge,
+  onViewDetails,
+  isExpanded,
+}) {
+  const { updateChallengeStatus } = useContext(ChallengesContext);
+
+  const formattedDate = new Date(challenge.deadline).toLocaleDateString(
```

Comment:
@sknepali Similar to the previous code, the formattedDate assumes that the challenge.deadline can be parsed into a valid date. If it can't, this could lead to unexpected behavior. We should add error handling to account for this.